File: frontend/src/services/inverterService.ts
import api from './api';
import type { Inverter } from '../types';

interface InverterApiItem {
  id?: number;
  code?: string;
  name?: string;
  section_id?: number;
  sectionId?: number;
  status?: string;
}

function mapInverter(item: InverterApiItem, index: number): Inverter {
  return {
    id: item.id ?? index + 1,
    sectionId: item.section_id ?? item.sectionId ?? 0,
    name: item.name ?? item.code ?? `Inverter ${index + 1}`,
    status: (item.status as Inverter['status']) ?? 'active',
    power: 0,
  } as Inverter;
}

export const inverterService = {
  // Inverters come from the backend /inverters endpoint. No mock inverters.
  getAll: async (): Promise<Inverter[]> => {
    const { data } = await api.get('/inverters');
    return (data?.data ?? []).map(mapInverter);
  },

  getBySection: async (sectionId: number): Promise<Inverter[]> => {
    const { data } = await api.get('/inverters');
    return (data?.data ?? [])
      .filter((inv: InverterApiItem) => (inv.section_id ?? inv.sectionId) === sectionId)
      .map(mapInverter);
  },

  getById: async (id: number): Promise<Inverter | undefined> => {
    const { data } = await api.get(`/inverters/${id}`);
    return data?.data ? mapInverter(data.data, 0) : undefined;
  },
};

File: backend/app/services/inverter_service.py
from typing import Any


class InverterService:
    async def get_all(self) -> dict[str, Any]:
        return {"status": "success", "message": "Inverter service ready", "inverters": []}

    async def get_by_id(self, inverter_id: int) -> dict[str, Any]:
        return {"status": "success", "message": f"Inverter {inverter_id} endpoint ready"}

    async def get_by_section(self, section_id: int) -> dict[str, Any]:
        return {"status": "success", "message": f"Inverters for section {section_id} endpoint ready"}

    async def create(self, data: dict) -> dict[str, Any]:
        return {"status": "success", "message": "Create inverter endpoint ready"}

    async def update(self, inverter_id: int, data: dict) -> dict[str, Any]:
        return {"status": "success", "message": f"Update inverter {inverter_id} endpoint ready"}

    async def delete(self, inverter_id: int) -> dict[str, Any]:
        return {"status": "success", "message": f"Delete inverter {inverter_id} endpoint ready"}
